### ISSUE TITLE
Incorrect value returned from vault for special characters.

### ISSUE THREAD

**Issue by the4yeast** [hashicorp/vault#2160]
This is for Vault v0.6.0

It's based on a real secret that I've stored in Vault. The real secret is obviously different, but it also starts with `%r`.

First, store in Vault:
`vault write secret/dummy value=%ra1`, response: `Success! Data written to: secret/dummy`.

Retrieve: `vault read secret/dummy`, response:
```
Key             	Value
---             	-----
refresh_interval	2592000
value           	%ra1
```

But, if we filter the output, `vault read -field=value secret/dummy`, the response is `%!r(MISSING)a1`.

There seems to be an issue around `%r` not escaped upon retrieval.

**Comment by jefferai** [comment]
Two questions:

1) Is this present in 0.6.3? 0.6 is quite old at this point.
2) Does the value look fine if you use format json?

**Comment by the4yeast** [comment]
1. Not sure about 0.6.3, my prod is using 0.6 and upgrading isn't really an option at the time. I stopped filtering output and I'm parsing it instead.
2. Looks fine with json and yaml, but when the whole thing is retrieved. Filtering by field results in the same problem, regardless of format.

**Comment by jefferai** [comment]
I think this is already fixed since 0.6.0:
```
$ vault read -field=value secret/dummy
%ra1
```

**Comment by the4yeast** [comment]
Thanks @jefferai, all I can do is upgrade
